_Week9/Group4/estimation.do

```diff
@@ -0,0 +1,99 @@
+********************************************************************************
+* ECON 491
+* Week 9 - Estimation
+* Group 4
+* November 14, 2022
+********************************************************************************
+
+cap prog drop projectdata_sim
+
+prog define projectdata_sim, rclass
+
+clear
+
+********************************************************************************
+* setting up inputs
+********************************************************************************
+
+{
+	
+syntax anything
+local bias: word 1 of `anything' //a dummy for whether the results are biased or not
+local cluster_size: word 2 of `anything' //inputting cluster size
+local number_of_schools: word 3 of `anything' //inputting number of schools
+local te: word 4 of `anything' //inputting treatment effect
+local takeup_rate: word 5 of `anything' //inputting takeup rate
+
+}
+
+********************************************************************************
+* data generating process
+********************************************************************************
+
+{
+
+clear
+set obs `number_of_schools' //setting number of schools based on input
+generate school_id = _n
+generate cluster_size = `cluster_size' //setting number of students to be observed at each school
+gen random = runiform(0,1)
+sort random
```

Comment:
Do you need to sort for the program to work? I didn't think so, but I could be wrong.